### PR TITLE
first and last name, how should we call

### PR DESCRIPTION
**Pull request by developer-rachit**
First and last name Textform field on enter_username.dart

### CODE REVIEW

File: lib/presentation/auth/widgets/enter_username.dart [hunk 1]
```diff
@@ -115,17 +187,33 @@ class _EnterUserNameState extends State<EnterUserName> {
     );
   }
 
-  String? _validate(String? value) {
+  String? _validateFirstName(String? value) {
```
Comment (Xazin) [inline lib/presentation/auth/widgets/enter_username.dart:190]: Generalize `_validateFirstName`  and `_validateLastName` - They're basically the same functions.

File: lib/presentation/auth/widgets/enter_username.dart [hunk 2]
```diff
@@ -57,36 +61,103 @@ class _EnterUserNameState extends State<EnterUserName> {
...
+                          RegExp("[a-zA-Z0-9_.-]")),
+                    ],
                   ),
-                ),
-                validator: _validate,
-                inputFormatters: <TextInputFormatter>[
-                  FilteringTextInputFormatter.allow(RegExp("[a-zA-Z0-9_.-]")),
                 ],
               ),
+              // child: TextFormField(
```
Comment (Xazin) [inline lib/presentation/auth/widgets/enter_username.dart:131]: Remove Commented Code

File: lib/presentation/auth/widgets/enter_username.dart [hunk 3]
```diff
@@ -15,9 +15,13 @@ class EnterUserName extends StatefulWidget {
 
 class _EnterUserNameState extends State<EnterUserName> {
   final _usernameController = TextEditingController();
```
Comment (Xazin) [inline lib/presentation/auth/widgets/enter_username.dart:17]: This variable is unused right?

File: lib/presentation/auth/widgets/enter_username.dart [hunk 4]
```diff
@@ -100,6 +171,7 @@ class _EnterUserNameState extends State<EnterUserName> {
                     onTap: () {
                       if (_isNameValid && state is! AwaitingUsernameUpdate) {
                         FocusScope.of(context).unfocus();
+                        _username = _firstName! + " " + _lastName!;
                         context
                             .read<AuthBloc>()
                             .add(AuthEvent.updateUsername(_username!));
```
Comment (Xazin) [inline lib/presentation/auth/widgets/enter_username.dart:177]: This needs to be changed, it needs to send firstName and lastName to the API separated, not as only firstName

File: lib/presentation/auth/widgets/enter_username.dart (hunk 3 above)
Comment (developer-rachit) [reply lib/presentation/auth/widgets/enter_username.dart:17]: Yes, it is removed now

File: lib/presentation/auth/widgets/enter_username.dart (hunk 2 above)
Comment (developer-rachit) [reply lib/presentation/auth/widgets/enter_username.dart:131]: Sure, removed

File: lib/presentation/auth/widgets/enter_username.dart (hunk 1 above)
Comment (developer-rachit) [reply lib/presentation/auth/widgets/enter_username.dart:190]: They have different String error messages, is there a way to identify TextFormField is it of Firstname or Lastname.

File: lib/presentation/auth/widgets/enter_username.dart (hunk 4 above)
Comment (developer-rachit) [reply lib/presentation/auth/widgets/enter_username.dart:177]: Kindly advise, `updateUsername(String username)` has only a single string parameter

File: lib/presentation/auth/widgets/enter_username.dart (hunk 1 above)
Comment (developer-rachit) [reply lib/presentation/auth/widgets/enter_username.dart]: To resolve added keys in Textfromfield, And then on the basis of keys display validation messages.

**Pull request closed without merging**